newIDE/app/src/MainFrame/index.js

```diff
@@ -517,7 +517,10 @@ export default class MainFrame extends React.Component<Props, State> {
     const { currentProject } = this.state;
     if (!currentProject) return;
 
-    if (!currentProject.hasLayoutNamed(oldName)) return;
```

Comment:
I think you reverse the check but that should not be the case.
The layout/external layout/events with the oldName MUST exist to be able to rename them, otherwise we bail out. It's a protection against a crash, in case for some reason the layouts were renamed somewhere else but the mainframe is not updated (which should not happen anyway).

What you want to check instead is: are there layout/external layout/events with the NEW name, and if yes, show the warning and bail out. It's done for objects I think

Reply:
So i do something like this for renameExternalLayout ?
```
   if (!currentProject.hasExternalLayoutNamed(oldName)) return;
    
    if (currentProject.hasExternalLayoutNamed(newName)) {
      showWarningBox('Another external layout with this name already exists.');
      return;
    };
```
And i adapt renameLayout, renameExternalEvents too ?